pkg/cache/v3/linear.go

```diff
@@ -218,6 +218,32 @@ func (cache *LinearCache) DeleteResource(name string) error {
 	return nil
 }
 
+// UpdateResources updates/deletes a list of resources in the cache.
+// Calling UpdateResources instead of iterating on UpdateResource and DeleteResource
```

Comment:
I have no problem updating the name of the method if it can create confusion with Update/Delete unique resources
But I prefer to keep a single call as it is more efficient on control-plane side, on the wire (one message instead of two) and imo it makes more sense functionally to update the cache only once

Reply:
Fair enough. Minimal calls is probably best anyways considering we want the speed increase. I think this is fine as `UpdateResources`